backends/gstreamer/media_capture.rs

```diff
@@ -8,37 +9,37 @@ use std::i32;
 
 trait AddToCaps {
     type Bound;
-    fn add_to_caps<'a>(
+    fn add_to_caps(
         &self,
         name: &str,
         min: Self::Bound,
         max: Self::Bound,
-        builder: gst::caps::Builder<'a>,
-    ) -> Option<gst::caps::Builder<'a>>;
+        builder: gst::caps::Builder<NoFeature>,
+    ) -> Option<gst::caps::Builder<NoFeature>>;
 }
 
 impl AddToCaps for Constrain<u32> {
     type Bound = u32;
-    fn add_to_caps<'a>(
+    fn add_to_caps(
         &self,
         name: &str,
         min: u32,
         max: u32,
-        builder: gst::caps::Builder<'a>,
-    ) -> Option<gst::caps::Builder<'a>> {
+        builder: gst::caps::Builder<NoFeature>,
+    ) -> Option<gst::caps::Builder<NoFeature>> {
         match self {
             Constrain::Value(v) => Some(builder.field(name, v)),
             Constrain::Range(r) => {
                 let min = into_i32(r.min.unwrap_or(min));
                 let max = into_i32(r.max.unwrap_or(max));
                 let range = gst::IntRange::<i32>::new(min, max);
-                if let Some(ideal) = r.ideal {
-                    let ideal = into_i32(ideal);
-                    let array = gst::List::new(&[&ideal, &range]);
-                    Some(builder.field(name, &array))
-                } else {
-                    Some(builder.field(name, &range))
-                }
+                Some(builder.field(name, &range))
+                // if let Some(ideal) = r.ideal {
```

Comment:
This probably needs fixing

Reply:
Do you know how to add this ideal plus a range? I was not sure how this looks like in the latest version of the GStreamer bindings.

Reply:
Well, the old code could've never worked, I think. You can't have a list that contains a range and a plain value.

The way how you could usually solve this is to create caps with two `gst::Structure`s. One has the ideal value, the other has the range, and the one with the ideal value comes first.

For that you could create first a `gst::Structure` with all fields and the range. And if there's an ideal value you would create a copy of that structure and set that one field to the ideal value. And then you could do e.g. `[structure_a, structure_b].into_iter().collect::<gst::Caps>()` or one of the many other ways (including `gst::Caps::builder_full()` and creating empty caps and `gst::CapsRef::append_structure()`).

Reply:
I was trying to make this change here, but this method is working on a `gst::caps::Builder<NoFeature>`, which seems to work only on a single structure. I guess this needs to be a refactor...

Reply:
Yes, also unrelated to the bindings version update. The previous code was simply broken :)

Reply:
Ok, I will change the commented code to a TODO comment and leave at it for now.